_drafts/restless-restfull.md

```diff
@@ -0,0 +1,776 @@
+you a lot of power. So I started reading the
+[Architectural Styles and the Design of Network-based Software Architectures](https://www.ics.uci.edu/~fielding/pubs/dissertation/top.htm)
+dissertation. AFAIK is the origin of the acronym REST.
+
+Since I was a little biased against RESTful stuff I was surprised
+at how awesome and different from everything you read about REST
+the dissertation was.
+
+The best part of it is the one that does not talk about REST, it actually
+builds a framework on how to compare different architectural styles
+to see which one fits better to your system. You can even create
+your own architectural style by composing existing architectural
+constraints, it was considerably enlightening to me and I highly
+recommend the read to anyone on the business of developing
+REST API's.
+
+How to properly architecture software starts on the dedication
+of the dissertation, it has an awesome quote from a book that
+is focused on architecture but it is very appliable on software
+architecture:
+
+```
+Almost everybody feels at peace with nature: listening to the
+ocean waves against the shore, by a still lake, in a field of grass,
+on a windblown heath.
+
+One day, when we have learned the timeless way again,
+we shall feel the same about our towns, and we shall feel as much at
+peace in them, as we do today walking by the ocean, or stretched out in
+the long grass of a meadow.
+
+-- Christopher Alexander, The Timeless Way of Building (1979)
+```
+
+This quote is a source of inspiration on what software architecture should
+be about. It is the problem that you are solving, the people that are
+going to use that architecture everyday (will live inside of it).
+
+Software has two audiences, other developers that have to evolve the
+system and clients that just use the system. The definition of a successful
+architecture is when both are extremely pleased, and being pleased comes
+through simplicity, just like nature, it simply fits and feels good.
+
+It is a social phenomenon, deeply influenced by human cognition. Because
+what is easy to understand and feels confortable is related to cognition.
+
+Of course that the architecture also has to support the problem being
+solved, no one likes a building that is confortable but may collapse
+on your head. That is why architecting is so hard, you have a lot
+of variables that will push your design on different directions,
+the decisions of the tradeoffs will be your responsibility.
+
+Uniformity is important, so discussing good practices and trying to
+enforce uniformity may be good, but uniformity has tradeoffs too, so
+you need to know when to give it up, the dissertation is also very clear
+on this.
+
+The dissertation is pretty big, but totally worth it, I can't quote
+everything I find interesting from it here because it would make
+this post even bigger than already it is (if you are curious
+you can see [my notes here](TODO), buy you will be best served
```

Comment:
https://github.com/katcipis/sophia/tree/master/notes

Reply:
yeah, thanks =D, I was lazy to get it when I was writing =P